spark/pom.xml

```diff
@@ -18,4 +18,60 @@
         <module>spark-version</module>
         <module>dist</module>
     </modules>
+
+    <dependencies>
+        <!--
+          This is a dummy dependency that is used to trigger the maven-shade plugin so that BigDL's
+          published POMs are flattened and do not contain variables. Without this dependency, some
+          subprojects' published POMs would contain variables like ${spark.version} that will
+          be substituted according to the default properties instead of the ones determined by the
+          profiles that were active during publishing. By ensuring that maven-shade runs for all
+          subprojects, we eliminate this problem because the substitutions are baked into the final POM.
+        -->
+        <dependency>
+            <groupId>org.spark-project.spark</groupId>
```

Comment:
Just 'borrow' the spark dummy project. Or we need to publish one to maven central. I think it's not so necessary. I will add more comments to explain this.